.github/workflows/build.yaml

```diff
@@ -10,43 +10,77 @@ on:
 
 jobs:
   build-binaries:
+    permissions:
+      id-token: write # for AWS OIDC
+
     strategy:
       matrix:
-        runs-on: [buildjet-4vcpu-ubuntu-2204, buildjet-4vcpu-ubuntu-2204-arm]
+        include:
+          - runner: buildjet-4vcpu-ubuntu-2204
+            arch: amd64
+          - runner: buildjet-4vcpu-ubuntu-2204-arm
+            arch: arm64
 
-    runs-on: ${{matrix.runs-on}}
+    runs-on: ${{matrix.runner}}
 
     steps:
       - uses: actions/checkout@v3
 
+      - name: Authenticate to AWS
+        if: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          aws-region: us-east-1
+          role-to-assume: arn:aws:iam::970625735569:role/GitHubActionsECRPush
+
+      - name: Configure AWS Docker Auth
+        if: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
+        uses: aws-actions/amazon-ecr-login@v1
+        with:
+          mask-password: true
+          registry-type: public
+
+      - name: Generate Image Metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            public.ecr.aws/edgebit/cluster-agent
+          tags: |
+            type=ref,event=pr,suffix=-${{ matrix.arch }}
```

Comment:
Note: upon merge, the SBOM did not have the image-tag set, I believe due to missing `event=branch`.

The action runs report warnings:
> [build-binaries (buildjet-4vcpu-ubuntu-2204, amd64)](https://github.com/edgebitio/cluster-agent/actions/runs/7301521532/job/19898317564#step:5:349)
> No Docker image version has been generated. Check tags input.